master/internal/api_experiment.go

```diff
@@ -2216,7 +2221,16 @@ func (a *apiServer) SearchExperiments(
 		Column("trials.experiment_id").
 		Column("trials.runner_state").
 		Column("trials.checkpoint_count").
-		Column("trials.summary_metrics").
+		ColumnExpr(`jsonb_build_object(
+				'avg_metrics',
+					(SELECT jsonb_object_agg(m.key, CASE WHEN m.value -> 'type' = '"number"'::jsonb
+						THEN m.value - '{sum, count}'::text[] ||
+							jsonb_build_object(
+								'mean', (m.value ->> 'sum')::float8 / (m.value ->> 'count')::int)
+						ELSE m.value END)
+					FROM jsonb_each(summary_metrics -> 'avg_metrics') AS m(key, value)),
+				'validation_metrics', summary_metrics -> 'validation_metrics'
+				) AS summary_metrics`).
```

Comment:
issue: now that we support [generic metrics](https://github.com/determined-ai/determined/pull/7012), summary metrics may have other fields besides training metrics (`avg_metrics`) and validation metrics (`validation_metrics`). so maybe we merge the jsons rather than try to rebuild them:

```suggestion
		ColumnExpr(`trials.summary_metrics || jsonb_build_object(
				'avg_metrics',
					(SELECT jsonb_object_agg(m.key, CASE WHEN m.value -> 'type' = '"number"'::jsonb
						THEN m.value - '{sum, count}'::text[] ||
							jsonb_build_object(
								'mean', (m.value ->> 'sum')::float8 / (m.value ->> 'count')::int)
						ELSE m.value END)
					FROM jsonb_each(summary_metrics -> 'avg_metrics') AS m(key, value))
			) AS summary_metrics`).
```

Reply:
come to think of it, this change does not compute the mean for the validation metrics, but perhaps it should.

then I'd say switching the database from storing `sum` to storing `mean` directly like Claire has suggested would be a better solution to this.

Reply:
The validation metrics aren't touched because the columns api and experiment search api don't surface the aggregates on the validation metrics object. I would want to handle that (+ migrating the summary metrics objects to storing mean instead of dynamically calculating it) in a separate PR.

cc @hkang1

Reply:
>I would want to handle that (+ migrating the summary metrics objects to storing mean instead of dynamically calculating it) in a separate PR.

ok. it seems to me most of this PR will be removed if/when we add generic metrics or switch to storing mean.